Bugsnag/Storage/BSGStorageMigratorV0V1.m

```diff
@@ -0,0 +1,85 @@
+//
+//  BSGStorageMigratorV0V1.m
+//  Bugsnag
+//
+//  Created by Karl Stenerud on 04.01.21.
+//  Copyright © 2021 Bugsnag Inc. All rights reserved.
+//
+
+#import "BSGStorageMigratorV0V1.h"
+
+#import "BSGFileLocations.h"
+#import "BugsnagLogger.h"
+
+static NSString *getCachesDir() {
+    NSArray *dirs = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES);
+    if ([dirs count] == 0) {
+        bsg_log_err(@"Could not locate cache directory path.");
+        return nil;
+    }
+
+    if ([dirs[0] length] == 0) {
+        bsg_log_err(@"Cache directory path is empty!");
+        return nil;
+    }
+    return dirs[0];
+}
+
+@implementation BSGStorageMigratorV0V1
+
++ (BOOL) migrate {
+    NSString *bundleName = [[NSBundle mainBundle] infoDictionary][@"CFBundleName"];
+    NSString *cachesDir = getCachesDir();
+    if(cachesDir == nil) {
+        bsg_log_err(@"Could not migrate v0 data to v1.");
+        return false;
+    }
+    BSGFileLocations *files = [BSGFileLocations v1];
+
+    NSDictionary *mappings = @{
+        @"bugsnag_handled_crash.txt": files.flagHandledCrash,
+        @"bugsnag/config.json": files.configuration,
+        @"bugsnag/metadata.json": files.metadata,
+        @"bugsnag/state.json": files.state,
+        @"bugsnag/state/system_state.json": files.systemState,
+        @"bugsnag/breadcrumbs": files.breadcrumbs,
+        @"bsg_kvstore": files.kvStore,
+        [@"Sessions" stringByAppendingPathComponent:bundleName]: files.sessions,
+        [@"KSCrashReports" stringByAppendingPathComponent:bundleName]: files.kscrashReports,
+    };
+
+    NSFileManager *fm = [NSFileManager defaultManager];
+    NSError *err = nil;
+    bool success = true;
+
+    for (NSString *key in mappings) {
+        NSString *srcPath = [cachesDir stringByAppendingPathComponent:key];
+        NSString *dstPath = mappings[key];
+        if ([fm fileExistsAtPath:srcPath]) {
+            if([fm fileExistsAtPath:dstPath]) {
+                if(![fm removeItemAtPath:dstPath error:&err]) {
+                    bsg_log_err(@"Could not remove %@: %@", dstPath, err);
+                }
+            }
+            if(![fm moveItemAtPath:srcPath toPath:dstPath error:&err]) {
+                bsg_log_err(@"Could not move %@ to %@: %@", srcPath, dstPath, err);
+                success = false;
+            }
+        }
+    }
+
+    for(NSString *path in @[
+        [cachesDir stringByAppendingPathComponent:@"bsg_kvstore"],
+        [cachesDir stringByAppendingPathComponent:@"bugsnag"],
+        [cachesDir stringByAppendingPathComponent:@"KSCrashReports"],
+        [cachesDir stringByAppendingPathComponent:@"Sessions"],
+                          ]) {
+        if(![fm removeItemAtPath:path error:&err]) {
+            bsg_log_err(@"Could not remove %@: %@", path, err);
```

Comment:
When I trigger the migrator using one of the example apps, I see this error for `bsg_kvstore`.

```
2021-01-11 08:53:48.256886+0000 Bugsnag Test App[70652:2466634] [Bugsnag] [ERROR] Could not remove /Users/nick/Library/Developer/CoreSimulator/Devices/5AFE2FCA-EB57-45D2-A705-42F81D4031F3/data/Containers/Data/Application/4AE2D40C-55E3-434B-9E78-9DEB080BF5FE/Library/Caches/bsg_kvstore: Error Domain=NSCocoaErrorDomain Code=4 "“bsg_kvstore” couldn’t be removed." UserInfo={NSUserStringVariant=(
    Remove
), NSFilePath=/Users/nick/Library/Developer/CoreSimulator/Devices/5AFE2FCA-EB57-45D2-A705-42F81D4031F3/data/Containers/Data/Application/4AE2D40C-55E3-434B-9E78-9DEB080BF5FE/Library/Caches/bsg_kvstore, NSUnderlyingError=0x600000b52280 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

I suggest checking the error code before logging:
```suggestion
            if (!([err.domain isEqual:NSCocoaErrorDomain] && err.code == NSFileNoSuchFileError)) {
                bsg_log_err(@"Could not remove %@: %@", path, err);
            }
```